scripts/reset_iks_api_key.sh

```diff
@@ -54,14 +54,25 @@ if [ "${reset}" == true ]; then
     if [ "$PRIVATE_ENV" = true ]; then
         if [ "$CLUSTER_ENDPOINT" == "private" ] || [ "$CLUSTER_ENDPOINT" == "default" ]; then
             RESET_URL="https://private.$REGION.containers.cloud.ibm.com/v1/keys"
-            curl -H "accept: application/json" -H "Authorization: $IAM_TOKEN" -H "X-Auth-Resource-Group: $RESOURCE_GROUP_ID" -X POST "$RESET_URL"
+            result=$(curl -i -H "accept: application/json" -H "Authorization: $IAM_TOKEN" -H "X-Auth-Resource-Group: $RESOURCE_GROUP_ID" -X POST "$RESET_URL" 2>/dev/null)
+            status_code=$(echo "$result" | head -n 1 | cut -d$' ' -f2)
```

Comment:
This doesn't seem very clean. `curl` has a built in way to fail if the response code is a failure response using the `--fail` flag. Take a look at https://curl.se/docs/manpage.html

Reply:
This is the current output after the change if the reset fails:
![Screenshot 2024-10-09 at 3 12 45 PM](https://github.com/user-attachments/assets/8e107f14-f241-49c7-867d-941543c29099)

This is the error when we use `--fail` and `--show-error`
![Screenshot 2024-10-09 at 3 32 11 PM](https://github.com/user-attachments/assets/0856b546-95c1-4b70-b8e7-9ee7098d62b5)

Reply:
ok this output looks good